sklego/decomposition/umap_reconstruction.py

```diff
@@ -44,6 +47,12 @@ class UMAPOutlierDetection(BaseEstimator, OutlierMixin):
         The offset used for the decision function.
     """
 
+    _parameter_constraints: dict = {
```

Comment:
From my understanding this is the official way of validating arguments. We are not there yet, but for `n_components` there was no other way of forcing param setting

Reply:
ahh this does feel a bit nasty though, do we really need to import `Interval` from a hidden module? If so, are we 100% sure that this is the way sklearn wants to expose these tools?